sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -58,8 +59,8 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
    * Resolve/create a primitive type.
    */
   override def visitPrimitiveDataType(ctx: PrimitiveDataTypeContext): DataType = withOrigin(ctx) {
-    val typeName = ctx.`type`.start.getType
-    (typeName, ctx.INTEGER_VALUE().asScala.toList) match {
+    val typeCtx = ctx.`type`
+    (typeCtx.start.getType, ctx.INTEGER_VALUE().asScala.toList) match {
```

Comment:
cc @hvanhovell @HyukjinKwon , since we put the SQL parser in the Spark Connect Scala client component, adding new data types means a client version upgrade.

Reply:
adding @hvanhovell 

We will have a meeting with Herman on Thursday to discuss how collations play with Spark Connect.

I just gave it a quick spin and seems like we lack collation support in spark connect even without this PR. e.g.

```
sql("select 'abc' collate 'UCS_BASIC_LCASE'")
```

gets us:

pyspark.errors.exceptions.connect.SparkConnectGrpcException: (org.apache.spark.sql.connect.common.InvalidPlanInput) Does not support convert string(UCS_BASIC_LCASE) to connect proto types.